### PR TITLE
Refactor the installation documentation

### LINKED ISSUE #362: Restructure MinIO Installation Pages
# Summary

Split out the current installation pages into sub pages:

```
Installation and Management
   -> Deploy MinIO in Distributed Mode
   -> Expand a Distributed MinIO Deployment
   -> Deploy MinIO in Standalone Mode
   -> Deploy MinIO in a Container
   -> Deploy MinIO in Kubernetes
   -> Upgrade a MinIO Deployment
```

# Goals

- Restructure existing content
- Improve quality of examples
- Include instructions related to embedded Console

# Non-Goals

- Create Production Guidance
- Create example distributed topologies

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

The current installation page could use some overhaul, especially in light of the new embedded MinIO Console. This pass also cleans up the structure to be a bit more guided, visible, and searchable. 

Also adding in some of the general production guidance we've put together for the K8s/VMware docs.

Closes #362 

# Goals

- Clean up structure of installation pages
- Create dedicated deployment + expansion tutorials for distributed MinIO
- Convert docker -> podman for containerized installations
- Link to MinIO Operator docs for K8s installations.

# Non-Goals

- Podman Distributed installations
- Podman + Docker Compose guides

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: source/installation/deploy-manage-minio-distributed.rst [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+   :width: 600px
+   :alt: MinIO Console Dashboard displaying Monitoring Data
+   :align: center
+
+You can use the MinIO Console for general administration tasks like
+Identity and Access Management, Metrics and Log Monitoring, or 
+Server Configuration. Each MinIO server includes its own embedded MinIO
+Console.
+
+Deployment Recommendations
```
Comment (ravindk89) [inline source/installation/deploy-manage-minio-distributed.rst:479]: @eco-minio this is mostly taken from the other published docs sites. LMK if you think anything needs tweaking.

File: source/installation/deploy-manage-minio-distributed.rst [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+--------
+
+A distributed MinIO deployment consists of 4 or more drives/volumes managed by
+one or more :mc:`minio server` process, where the processes manage pooling the
+compute and storage resources into a single aggregated object storage resource.
+Each MinIO server has a complete picture of the distributed topology, such that
+an application can connect to any node in the deployment and perform S3
+operations.
+
+Distributed deployments enable implicitly enable :ref:`erasure coding
```
Comment (kannappanr) [inline source/installation/deploy-manage-minio-distributed.rst:23]: `Distributed deployments enable implicitly` -> `Distributed deployments implicitly`

File: source/installation/deploy-manage-minio-distributed.rst [hunk 3]
```diff
@@ -0,0 +1,620 @@
...
+Networking and Hostnames
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each node should have full bidirectional network access to every other
+node in the deployment. For containerized or orchestrated infrastructures,
+this may require specific configuration of networking and routing 
+components such as ingress or load balancers.
+
+MinIO *requires* using sequentially-numbered hostnames to represent each
+:mc:`minio server` process in the deployment. Create the necessary DNS hostname
```
Comment (kannappanr) [inline source/installation/deploy-manage-minio-distributed.rst:55]: Is the reference here correct?

File: source/installation/deploy-manage-minio-distributed.rst [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+
+Erasure Coding SLA
+   MinIO requires each pool satisfy the deployment
+   :ref:`erasure code <minio-erasure-coding>` settings. Specifically,
+   the new pool must have *at least* 2x the number of drives as the
+   :ref:`Standard <minio-ec-storage-class>` parity storage class.
+
+   For example, consider a MinIO deployment with a single 16-node server pool
+   using the default erasure code parity ``EC:4``. The new server pool
+   must have enough drives to satisfy ``EC:8``.
```
Comment (kannappanr) [inline source/installation/deploy-manage-minio-distributed.rst:285]: The new server pool
   must have enough drives to satisfy ``EC:4``

File: source/installation/deployment-and-management.rst [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+
+Applications should use the ``https://HOST-ADDRESS:9000`` to perform S3
+operations against the MinIO server.
+
+.. _deploy-minio-standalone-container:
+
+Deploy Standalone MinIO in a Container
+--------------------------------------
+
+The following procedure deploys a single MinIO container in with single drive or
```
Comment (kannappanr) [inline source/installation/deployment-and-management.rst:231]: `The following procedure deploys a single MinIO container in with single drive` -> `The following procedure deploys a single MinIO container with single drive`

File: source/installation/deployment-and-management.rst [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+
+Create a file ``config.env`` using your preferred text editor and enter the
+following environment variables:
+
+.. code-block:: shell
+   :class: copyable
+
+   export MINIO_ROOT_USER=minio-admin
+   export MINIO_ROOT_PASSWORD=minio-secret-key-CHANGE-ME
+   export MINIO_KMS_SECRET_KEY=my-minio-encryption-key:bXltaW5pb2VuY3J5cHRpb25rZXljaGFuZ2VtZTEyMwo=
```
Comment (kannappanr) [inline source/installation/deployment-and-management.rst:255]: Should this be documented? It is better to use KES.

**Review by ravindk89 — COMMENTED** [review]
@kannappanr addressed comments.

**Pull request merged**